nurseconnect/views.py

```diff
@@ -50,46 +60,266 @@ def search(request, results_per_page=10):
 
 
 class RegistrationView(FormView):
+    """
+    Handles user registration
+    """
     form_class = forms.RegistrationForm
     template_name = "profiles/register.html"
 
     def form_valid(self, form):
         username = form.cleaned_data["username"]
         password = form.cleaned_data["password"]
-        user = User.objects.create_user(username=username, password=password)
-        user.save()
-
-        authed_user = authenticate(username=username, password=password)
-        login(self.request, authed_user)
-        return HttpResponseRedirect(reverse("home"))
+        try:
+            user = User.objects.create_user(
+                username=username,
+                password=password
+            )
+            user.save()
+            # TODO: save security questions
+            for index, question in enumerate(models.SecurityQuestion.objects.all()):
+                answer = form.cleaned_data["question_%s" % index]
+                models.SecurityAnswer.objects.create(
+                    user=user.profile,
+                    question=question,
+                    answer=answer
+                )
+            authed_user = authenticate(username=username, password=password)
+            login(self.request, authed_user)
+            return HttpResponseRedirect(reverse("home"))
+        except User.DoesNotExist:
+            form.add_error("username",
+                           _("The username that you entered appears to be "
+                             "invalid. Please try again."))
+            return self.render_to_response({"form": form})
 
     def render_to_response(self, context, **response_kwargs):
         return super(RegistrationView, self).render_to_response(
             context, **response_kwargs
         )
 
+    def get_form_kwargs(self):
+        kwargs = super(RegistrationView, self).get_form_kwargs()
+        kwargs["questions"] = models.SecurityQuestion.objects.all()
+        return kwargs
+
+
+class MyProfileView(View):
+    template_name = "profiles/viewprofile.html"
+    settings_form = forms.EditProfileForm(prefix="settings_form")
+    profile_password_change_form = forms.ProfilePasswordChangeForm(
+        prefix="profile_password_change_form"
+    )
 
-class EditProfileView(FormView):
-    model = User
-    form_class = forms.EditProfileForm
-    template_name = "profiles/editprofile.html"
-    success_url = reverse_lazy("molo.profiles:view_my_profile")
+    def get(self, request, *args, **kwargs):
```

Comment:
This is all related to `context`.
Might be better if we have this in `def get_context_data(self, **kwargs):`